modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/SharingUrlTimestampParser.kt

```diff
@@ -0,0 +1,37 @@
+package au.com.shiftyjelly.pocketcasts.utils
+
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+class SharingUrlTimestampParser {
+    fun parseTimestamp(t: String): Pair<Duration?, Duration?> {
+        val intervalPattern = Regex("""(\d*\.?\d*)?,?(\d*\.?\d*)?""") // t=10,20 or t=,20 or t=10
+        val hmsPattern = Regex("""(?:(\d+)h)?(?:(\d+)m)?(?:(\d+(?:\.\d+)?)s)?""") // t=10h20m30s or t=10m30s or or 10m or t=10s
+        val hhmmssFractionPattern = Regex("""(\d+):(\d{2}):(\d{2})(?:\.(\d+))?""") // t=10:20:30.456
```

Comment:
Can we move this to companion object? Creating regex is a bit heavy operation because it needs to create the whole internal state machine.

Reply:
Done: [5c8e07d](https://github.com/Automattic/pocket-casts-android/pull/2235/commits/5c8e07dcb433aa4d3aafc07d7a38a30e0b6d7ef5)